client/index.js

```diff
@@ -47,7 +47,7 @@ function log(level, msg) {
 
 // Send messages to the outside, so plugins can consume it.
 function sendMsg(type, data) {
-	if(typeof self !== "undefined" && self.window) {
+	if(typeof self !== "undefined" && typeof WorkerGlobalScope === "undefined") {
```

Comment:
I think the correct check should be this:

```js
typeof self !== "undefined" &&
(typeof WorkerGlobalScope === "undefined" ||
!(self instanceof self.WorkerGlobalScope))
```

Use `instanceof` only if it not undefined. I changed it. :)